themes/src/main/resources/theme/base/login/webauthn-register.ftl

```diff
@@ -21,158 +21,10 @@
-                            }
-                        } else {
-                            console.log("Your browser is not able to recognize supported transport media for the authenticator.");
-                        }
-
-                        let initLabel = "WebAuthn Authenticator (Default Label)";
-                        let labelResult = window.prompt("Please input your registered authenticator's label", initLabel);
-                        if (labelResult === null) labelResult = initLabel;
-                        $("#authenticatorLabel").val(labelResult);
-
-                        $("#register").submit();
-
-                    })
-                    .catch(function (err) {
-                        $("#error").val(err);
-                        $("#register").submit();
-
-                    });
-            }
-
-            function getPubKeyCredParams(signatureAlgorithmsList) {
-                let pubKeyCredParams = [];
-                if (signatureAlgorithmsList === []) {
-                    pubKeyCredParams.push({type: "public-key", alg: -7});
-                    return pubKeyCredParams;
-                }
-
-                for (let i = 0; i < signatureAlgorithmsList.length; i++) {
-                    pubKeyCredParams.push({
-                        type: "public-key",
-                        alg: signatureAlgorithmsList[i]
-                    });
-                }
-                return pubKeyCredParams;
-            }
-
-            function getExcludeCredentials(excludeCredentialIds) {
-                let excludeCredentials = [];
-                if (excludeCredentialIds === "") return excludeCredentials;
-
-                let excludeCredentialIdsList = excludeCredentialIds.split(',');
-
-                for (let i = 0; i < excludeCredentialIdsList.length; i++) {
-                    excludeCredentials.push({
-                        type: "public-key",
-                        id: base64url.decode(excludeCredentialIdsList[i],
-                        {loose: true})
-                    });
-                }
-                return excludeCredentials;
-            }
-
-            function getTransportsAsString(transportsList) {
-                if (transportsList === '' || transportsList.constructor !== Array) return "";
-
-                let transportsString = "";
-
-                for (let i = 0; i < transportsList.length; i++) {
-                    transportsString += transportsList[i] + ",";
-                }
-
-                return transportsString.slice(0, -1);
+                registerByWebAuthn('${challenge}', '${userid}', '${username}', [<#list signatureAlgorithms as sigAlg>${sigAlg},</#list>], '${rpEntityName}', '${rpId}', '${attestationConveyancePreference}', '${authenticatorAttachment}', '${requireResidentKey}', '${userVerificationRequirement}', ${createTimeout}, '${excludeCredentialIds}', '${msg("webauthn-unsupported-browser-text")?no_esc}');
```

Comment:
Rather than passing such a large amount of arguments, it might be preferential to pass this as an object with keys and values, it will make it less error prone to refactor in the future in case arguments shift.

Reply:
I fixed the codes as you suggested.